### PR TITLE
feat: implement restore table to a different instance feature (#830)

### PR DESCRIPTION
**Pull request by kolea2**
* feat: implement restore table to a different instance feature

* chore: docs

* chore: add system test

* fix(samples): region tag

* fix: docs

### CODE REVIEW

**Comment by kolea2** [comment]
@liubonan fyi

**Review by craiglabenz — APPROVED** [review]
Lots of BigTable magic I can't say much about, but this looks good to me on a Node front. 

One interesting observation around loading instances from... an Id? To get back to its `name`? That was a bit unclear.

File: src/backup.ts [hunk 1]
```diff
@@ -491,10 +498,64 @@ Please use the format 'my-backup' or '${cluster.name}/backups/my-backup'.`);
...
+  restoreTo(
+    config: RestoreTableConfig,
+    callback?: RestoreTableCallback
+  ): void | Promise<RestoreTableResponse> {
+    let parent: string;
+    if (config.instance) {
+      if (config.instance instanceof Instance) {
+        parent = config.instance.name;
+      } else {
+        parent = this.bigtable.instance(config.instance).name;
```
[the remark this answers is not in the document]
Comment (kolea2) [reply src/backup.ts:553]: will take a look at this, thanks!

[the remark this answers is not in the document]
Comment (kolea2) [reply src/backup.ts:553]: Opened https://github.com/googleapis/nodejs-bigtable/issues/870 to track this refactor

**Review by crwilcox — APPROVED** [review]

File: src/backup.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (stephenplusplus) [reply src/backup.ts:553]: The Instance constructor determines the format of the name given and adjusts (if necessary) the full format for the `name` property: https://github.com/googleapis/nodejs-bigtable/blob/bac490d35b82b1aeea99aba5a117b5ec586c4b39/src/instance.ts#L177-L190

We could duplicate that logic here, or maybe move it to a utility function. However, I don't believe there is a big cost to creating the instance here, and letting the g/c pick it up.

[the remark this answers is not in the document]
Comment (kolea2) [reply src/backup.ts:553]: I think this is ok to leave as is for now, will resolve this.

**Pull request merged**